src/shared/components/iframe-runtime.tsx

```diff
@@ -63,8 +64,12 @@ export const IframeRuntime: React.FC<IProps> =
         if (options.unloading) {
           return new Promise<any>(resolve => {
             resolveOnUnload.current = resolve;
-            // send the unload request to the wrapped interactive
-            phone.post("getInteractiveState", options);
+            if (!isSubinteractive) {
```

Comment:
This approach isn't great, but something like it might be the best quick fix.

It isn't great because it means there might be lost work when the student changes the page immediately after making a change in the sub interactive. 

A better, but harder fix, would be for the hosting interactive to provide the "setOnUnload" function as a prop to the iframe-runtime. Then the hosting interactives can take the result from `resolveOnUnload.current?.(newInteractiveState);` and wrap it the same way they are wrapping the interactive state of their sub interactives normally.

Reply:
cc @dougmartin

Reply:
@scytacki Thanks. Yeah, I kind of suspected it wouldn't be this easy or it would have been done already. I put this up so I could show Doug when I asked him (on Slack) if it'd be a bad idea.

I'll look into the approach you suggested.